### PR TITLE
Allow querying AzureAISearch without non-null metadata field

### PR DESCRIPTION
**Pull request by chrispangg**
# Description

In our project we are indexing data with an Azure AI Search Indexer. As it's not a requirement use metadata in the project, we faced an issue when we tried to query the index with LlamaIndex. In the current codebase, it is a requirement to have a non-null metadata field. To fix this issue, we created an empty metadata field in Azure. However, this is not enough to resolve the issue as the field is still null.

We want to use LlamaIndex to query the index without committing to adding metadata to our documents.

In the PR, I am bypassing this by passing in an empty dict if metadata is null. I have tested this locally and it's working for us.

Fixes # (issue)

## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Added new unit/integration tests
- [ ] Added new notebook (that tests end-to-end)
- [x] I stared at the code and made sure it makes sense

## Suggested Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added Google Colab support for the newly added notebooks.
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I ran `make format; make lint` to appease the lint gods

### CODE REVIEW

**Review by nerdai — COMMENTED** [review]
sgtm @chrispangg! Thanks for this. Please also bump the version number in the package's pyproject.toml.

File: pyproject.toml
Comment (nerdai) [inline pyproject.toml:1]: Sorry I should've been more specific. Not this one but the pyproject.toml file within the azureaisearch vector store integration.

**Comment by chrispangg** [comment]
Version bumped now 👍 Thanks @nerdai!

**Review by nerdai — COMMENTED** [review]
Sorry that I wasn't clear before. But can you please revert the changes to the pyproject.toml in the root llama-index folder and adjust the one in the azureaisearch folder instead?

**Review by logan-markewich — APPROVED** [review]

**Pull request merged**